### PR TITLE
GOVUKAPP-996: Disable backups

### PR DESCRIPTION
**Pull request by davidc-gds**
# Title

Disable backups

## JIRA ticket(s)
  - [GOVUKAPP-996](https://govukverify.atlassian.net/browse/GOVUKAPP-996)

[GOVUKAPP-996]: https://govukverify.atlassian.net/browse/GOVUKAPP-996?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -4,9 +4,9 @@
 
     <application
         android:name=".GovUkApplication"
-        android:allowBackup="true"
+        android:allowBackup="false"
```
Comment (TimGJD) [inline app/src/main/AndroidManifest.xml:7]: The guidance I got from Jon and Ryan for this ticket was for the following:

- Allow backups
- Exclude db data
- Include shared prefs like iOS (as apparently iOS couldn't exclude them)

?

File: app/src/main/res/xml/data_extraction_rules.xml [hunk 2]
```diff
@@ -1,5 +1,25 @@
...
+        <exclude domain="device_root" />
+        <exclude domain="device_file" />
+        <exclude domain="device_database" />
+        <exclude domain="device_sharedpref" />
     </cloud-backup>
+    <device-transfer>
+        <exclude domain="root" />
+        <exclude domain="file" />
+        <exclude domain="database" />
+        <exclude domain="sharedpref" />
```
Comment (TimGJD) [inline app/src/main/res/xml/data_extraction_rules.xml:18]: See comment on manifest

File: app/src/main/res/xml/data_extraction_rules.xml [hunk 3]
```diff
@@ -1,5 +1,25 @@
...
+        <exclude domain="external" />
+        <exclude domain="device_root" />
+        <exclude domain="device_file" />
+        <exclude domain="device_database" />
+        <exclude domain="device_sharedpref" />
```
Comment (TimGJD) [inline app/src/main/res/xml/data_extraction_rules.xml:23]: See comment on manifest

File: app/src/main/res/xml/data_extraction_rules.xml [hunk 4]
```diff
@@ -1,5 +1,25 @@
...
     <cloud-backup>
+        <exclude domain="root" />
+        <exclude domain="file" />
+        <exclude domain="database" />
+        <exclude domain="sharedpref" />
+        <exclude domain="external" />
+        <exclude domain="device_root" />
+        <exclude domain="device_file" />
+        <exclude domain="device_database" />
+        <exclude domain="device_sharedpref" />
```
Comment (TimGJD) [inline app/src/main/res/xml/data_extraction_rules.xml:12]: See comment on manifest

File: app/src/main/res/xml/data_extraction_rules.xml [hunk 5]
```diff
@@ -1,5 +1,25 @@
 <?xml version="1.0" encoding="utf-8"?>
 <data-extraction-rules>
     <cloud-backup>
+        <exclude domain="root" />
+        <exclude domain="file" />
+        <exclude domain="database" />
+        <exclude domain="sharedpref" />
```
Comment (TimGJD) [inline app/src/main/res/xml/data_extraction_rules.xml:7]: See comment on manifest

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (davidc-gds) [reply app/src/main/AndroidManifest.xml:7]: I think that will lead to unexpected behaviour as the app will think the user has completed topic selection (as the flag is stored in shared prefs) but there will be no items in the database. Wouldn't it make more sense to start them with a clean slate?

Comment (davidc-gds) [reply app/src/main/AndroidManifest.xml:7]: @TimGJD Had a chat with Ryan and he has agreed that it makes sense to exclude everything for now.

**Review by gclssvglx — APPROVED** [review]

**Pull request merged**